Json formatting

Nightsd01 opened the pull request:
• Previously, when calling `postNotification()` or `getPermissionSubscriptionState()`, or when building errors, the Java wrapper was building JSON objects by converting from strings.
• This left the potential to have invalid JSON, which became apparent when we switched to FCM. The FCM push tokens have a colon in them which caused failures when parsing the JSON strings.
• Switched to using JSONObject's builder methods, which handles serializing invalid characters and null values.

Nightsd01 committed:
Invalid JSON Formatting

• Fixes an issue where the Java android wrapper was building JSON objects using strings, but wasn't properly formatting the JSON and would throw a JSONException
• This issue would have effected both getPermissionSubscriptionState() and postNotification()
• Fixed by using Java JSON's built in builder methods

Nightsd01 committed:
Error JSON

• Instead of serializing error JSON objects by converting from a string, switched to building error JSON objects using the put() method in case the error message is null/invalid JSON

jkasten2 commented on android/src/main/java/com/geektime/rnonesignalandroid/RNOneSignal.java:
```diff
@@ -95,6 +96,12 @@ private void sendEvent(String eventName, Object params) {
    }
 
+   private JSONObject jsonFromErrorMessageString(String errorMessage) throws JSONException {
+      JSONObject errorObject = new JSONObject();
```
Can be simplified to
```java
return new JSONObject().put("error", errorMessage);
```

Nightsd01 committed:
Update RNOneSignal.java

• Slightly simplified the method to construct an error object based off of an error message string

Nightsd01 committed:
Update RNOneSignal.java
